databroker/broker.py

```diff
@@ -690,15 +698,23 @@ def get_events(self, headers, fields=None, stream_name=ALL, fill=False,
         check_fields_exist(fields if fields else [], headers)
 
         for h in headers:
+            if not isinstance(h, Header):
+                h = self[h['start']['uid']]
+
+            descs = h.descriptors
             for es in self.event_sources:
                 gen = es.docs_given_header(
                         header=h, stream_name=stream_name,
-                        fill=fill,
-                        fields=fields,
-                        handler_registry=handler_registry,
-                        handler_overrides=handler_overrides)
-                for nm, ev in gen:
-                    if nm == 'event':
+                        fill=False,
+                        fields=fields)
+                gen = (ev for nm, ev in gen if nm == 'event')
+                # dirty hack!
+                es = self.assets['']
```

Comment:
Is there are reason to reuse `es` here? Isn't this a Registry, not an event source?